src/library/scala/Option.scala

```diff
@@ -171,7 +171,7 @@ sealed abstract class Option[+A] extends IterableOnce[A] with Product with Seria
    */
   final def isDefined: Boolean = !isEmpty
 
-  final def knownSize: Int = if (isEmpty) 0 else 1
+  override final def knownSize: Int = if (isEmpty) 0 else 1
```

Comment:
Shouldn't this be `if (isEmpty) 0 else -1`? Also arguably the default should just be `-1` because it may be expensive to compute `isEmpty` right?